src/strands/agent/agent.py

```diff
@@ -569,14 +596,19 @@ async def stream_async(
         """
         callback_handler = kwargs.get("callback_handler", self.callback_handler)
 
+        # runtime override or agent default TODO in the future, when we expose 'output_schema, we should consider allowing for halfway configuration. for example, the user should be able to define `output_mode` on the Agent level but `structured_output_type` on the `output_mode`
```

Comment:
This could use some elaboration in the description; not sure what is being said here